### ISSUE TITLE
[1.20.1] Skript not loading any sounds

### ISSUE THREAD

**Issue by Rijen147** [Teneted/Tenet#3435]
**Minecraft Version :** 1.20.1

**Mohist Version :** 1.20.1-896

**Operating System :** Windows

**Concerned mod / plugin** : Skript

**Logs :** 
![Discord_ORXzmbILYU](https://github.com/user-attachments/assets/8d0d59a3-6ac5-4c7b-bb9d-4fb1da5e1e3c)

**Steps to Reproduce :**
 1. Setup Skript in plugins
 2. Start any script with 'play sound' effect
 3. Get an error, that breaks the entire script
 
**Description of issue :** I've switched from ketting to mohist cause of paper support, I loaded all the scripts, Skript didn't send any inner errors. Whenever script is triggering an event with 'play sound' effect in it, it sends an error.

**Comment by Unitei-Dev** [comment]
> **Minecraft Version :** 1.20.1
> 
> **Mohist Version :** 1.20.1-896
> 
> **Operating System :** Windows
> 
> **Concerned mod / plugin** : Skript
> 
> **Logs :** ![Discord_ORXzmbILYU](https://private-user-images.githubusercontent.com/67170324/389103365-8d0d59a3-6ac5-4c7b-bb9d-4fb1da5e1e3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI0MTY1MjIsIm5iZiI6MTczMjQxNjIyMiwicGF0aCI6Ii82NzE3MDMyNC8zODkxMDMzNjUtOGQwZDU5YTMtNmFjNS00YzdiLWJiOWQtNGZiMWRhNWUxZTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI0VDAyNDM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4Yjg3Yjk1MTM4MTJmZWQ2NDA0NDk0ZjQwYzNmOTI2Nzc4MTc2NWZmMDg1MmYzMGI2ZWFkNzIxNjQ1MzQ2ODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ISjy-fgR3_b0QPCRDTnL6txSxqm3af1VMt68XNFOzMU)
> 
> **Steps to Reproduce :**
> 
> 1. Setup Skript in plugins
> 2. Start any script with 'play sound' effect
> 3. Get an error, that breaks the entire script
> 
> **Description of issue :** I've switched from ketting to mohist cause of paper support, I loaded all the scripts, Skript didn't send any inner errors. Whenever script is triggering an event with 'play sound' effect in it, it sends an error.

You are not using the latest version and the Script does not work well with mods use Denizen

**Comment by Unitei-Dev** [comment]
![image](https://github.com/user-attachments/assets/d3bfccd1-f8fc-48b7-81a7-85528dc1df58)

**Comment by SNWCreations** [comment]
This happens due to incomplete Paper API implementation. Maybe could be solved by #3443 by porting Sound-related Adventure API implementations from Paper.

**Comment by Rijen147** [comment]
> > **Minecraft Version :** 1.20.1
> > **Mohist Version :** 1.20.1-896
> > **Operating System :** Windows
> > **Concerned mod / plugin** : Skript
> > **Logs :** ![Discord_ORXzmbILYU](https://private-user-images.githubusercontent.com/67170324/389103365-8d0d59a3-6ac5-4c7b-bb9d-4fb1da5e1e3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI0MTY1MjIsIm5iZiI6MTczMjQxNjIyMiwicGF0aCI6Ii82NzE3MDMyNC8zODkxMDMzNjUtOGQwZDU5YTMtNmFjNS00YzdiLWJiOWQtNGZiMWRhNWUxZTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI0VDAyNDM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4Yjg3Yjk1MTM4MTJmZWQ2NDA0NDk0ZjQwYzNmOTI2Nzc4MTc2NWZmMDg1MmYzMGI2ZWFkNzIxNjQ1MzQ2ODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ISjy-fgR3_b0QPCRDTnL6txSxqm3af1VMt68XNFOzMU)
> > **Steps to Reproduce :**
> > 
> > 1. Setup Skript in plugins
> > 2. Start any script with 'play sound' effect
> > 3. Get an error, that breaks the entire script
> > 
> > **Description of issue :** I've switched from ketting to mohist cause of paper support, I loaded all the scripts, Skript didn't send any inner errors. Whenever script is triggering an event with 'play sound' effect in it, it sends an error.
> 
> You are not using the latest version and the Script does not work well with mods use Denizen

I'm not switching to another scripting language cuz of one problem, especially when it is caused by mohist not mods, please stop recommending me to do that. Also I've already tried newest version of Skript, error is not disappearing anywhere

**Comment by Rijen147** [comment]
> This happens due to incomplete Paper API implementation. Maybe could be solved by #3443 by porting Sound-related Adventure API implementations from Paper.

Thanks. How do I do that or will it be solved by mohist updates?

**Comment by SNWCreations** [comment]
may will be solved by updates.

**Comment by Unitei-Dev** [comment]
> > Это происходит из-за неполной реализации API Paper. Возможно, это можно решить с помощью #3443 , перенеся реализации API Sound-related Adventure из Paper.
> 
> Спасибо. Как мне это сделать или это будет решено обновлениями mohist?

Mohist not support Paper API)

**Comment by SNWCreations** [comment]
actually it supports but not fully supported, so the mentioned PR is tried to introduce Paper's change which related to Adventure impl.

**Comment by Unitei-Dev** [comment]
> actually it supports but not fully supported, so the mentioned PR is tried to introduce Paper's change which related to Adventure impl.

It no longer supports it, since it interferes with mods, you can do better in KubeJS than in plugins on Paper (I don't need them)

**Comment by SNWCreations** [comment]
that depends on user ¯\_(ツ)_/¯

**Comment by Unitei-Dev** [comment]
> that depends on user ¯_(ツ)_/¯

It depends on the user's development in Java, if it is not there, then there is no point in making a server without an encoder

**Comment by SNWCreations** [comment]
ye. so leaving this there and looking forwards to Mgazul's words on #3443 updates.

**Comment by Unitei-Dev** [comment]
> ye. so leaving this there and looking forwards to Mgazul's words on #3443 updates.

It's easier to create your own core with Paper + NeoForge, than I have coders now and have been jammed for the last half a year at 1.21.x 

Well, you can also create mods and plugins from your own API

**Comment by SNWCreations** [comment]
> > ye. so leaving this there and looking forwards to Mgazul's words on #3443 updates.
> 
> It's easier to create your own core with Paper + NeoForge, than I have coders now and have been jammed for the last half a year at 1.21.x
> 
> Well, you can also create mods and plugins from your own API

yeah... that's off-topic there as it is nearly impossible to ask a modded user to switch MC version.

Going to work so I'm stopping chat there. :/ twenty min to 7AM for UTC+8
